src/services/epg/epgClient.service.ts

```diff
@@ -18,25 +17,16 @@ export const isFulfilled = <T>(input: PromiseSettledResult<T>): input is Promise
   return false;
 };
 
-const epgProgramSchema = object().shape({
-  id: string().required(),
-  title: string().required(),
-  startTime: string()
-    .required()
-    .test((value) => (value ? isValid(new Date(value)) : false)),
-  endTime: string()
-    .required()
-    .test((value) => (value ? isValid(new Date(value)) : false)),
-  chapterPointCustomProperties: array().of(
-    object().shape({
-      key: string().required(),
-      value: string().test('required-but-empty', 'value is required', (value: unknown) => typeof value === 'string'),
-    }),
-  ),
-});
-
 @injectable()
-export default class EpgService {
+export default class EpgClientService {
+  private viewNexaProvider: EpgProviderService;
+  private jwProvider: EpgProviderService;
+
+  public constructor(@targetName(EPG_TYPE.JW) jwProvider: EpgProviderService, @targetName(EPG_TYPE.VIEW_NEXA) viewNexaProvider: EpgProviderService) {
```

Comment:
I think this is the perfect use case for getting multiple modules by type of EpgProviderService. This class can just get all of the EpgProviderService's that are defined and put them into a dictionary by 'type' (assuming we require each one to define a 'type' property.)

Reply:
That should work, but then we do want to prevent any imports/initialization from ending up in the constructor. For example, loading/importing a ViewNexa SDK.